api/Controllers/RoomController.cs

```diff
@@ -76,10 +78,12 @@ public async Task<IActionResult> GetRooms([FromRoute] int hotelId)
             {
                 return BadRequest("Such hotel does not exist");
             }
-            Room[] rooms = await _hotelsDb.Rooms
-                .Where(r => r.HotelId == hotelId)
-                .ProjectTo<Room>(_mapper.ConfigurationProvider)
-                .ToArrayAsync();
+            List<Room> rooms = await _hotelsDb.Hotels
+                .Where(h => h.Id == hotelId)
+                .Include(hotel => hotel.Rooms)
+                .ThenInclude(room => room.Facilities)
+                .Select(hotel => _mapper.Map<List<Room>>(hotel.Rooms))
+                .SingleOrDefaultAsync();
```

Comment:
Why did you start from Hotels `DbSet`?

Reply:
It is a good question, I should probably start from entity I need (room in this case), not from entity (hotel), which contains entity I need